rust/main/chains/hyperlane-cosmos/src/utils.rs

```diff
@@ -24,20 +24,25 @@ pub(crate) const CONTRACT_ADDRESS_ATTRIBUTE_KEY: &str = "_contract_address";
 pub(crate) static CONTRACT_ADDRESS_ATTRIBUTE_KEY_BASE64: Lazy<String> =
     Lazy::new(|| BASE64.encode(CONTRACT_ADDRESS_ATTRIBUTE_KEY));
 
-/// Given a lag, returns the block height at the moment.
-/// If the lag is None, a block height of None is given, indicating that the
-/// tip directly can be used.
-pub(crate) async fn get_block_height_for_lag(
+/// Given a `reorg_period`, returns the block height at the moment.
+/// If the `reorg_period` is None, a block height of None is given,
+/// indicating that the tip directly can be used.
+pub(crate) async fn get_block_height_for_reorg_period(
     provider: &WasmGrpcProvider,
-    lag: Option<NonZeroU64>,
+    reorg_period: &ReorgPeriod,
 ) -> ChainResult<Option<u64>> {
-    let block_height = match lag {
-        Some(lag) => {
+    let block_height = match reorg_period {
+        ReorgPeriod::Blocks(blocks) => {
             let tip = provider.latest_block_height().await?;
-            let block_height = tip - lag.get();
+            let block_height = tip - blocks.get() as u64;
             Some(block_height)
         }
-        None => None,
+        ReorgPeriod::None => None,
+        ReorgPeriod::Tag(_) => {
+            return Err(ChainCommunicationError::CustomError(
+                "Cosmos does not support reorg period as a tag".into(),
```

Comment:
would be useful to add an entry to [HyperlaneCosmosError](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/02a5b92ba774c3ce8c706966c9b9f23272215fa5/rust/main/chains/hyperlane-cosmos/src/error.rs#L13) and have a string parameter inside, to include the `tag` in the error message for easier debugging. Then here you can return e.g. `HyperlaneCosmosError::ReorgPeriodValueNotSupported(tag).into()`

Reply:
Good idea! Maybe it would be even better to add it to `ChainCommunicationError`, something like
```
#[error("Unsupported reorg period: {0:?}")]
UnsupportedReorgPeriod(ReorgPeriod),
```
and reuse it in `EthereumReorgPeriod` too?